### ISSUE TITLE
[Bug]: block explorer `index` does not always match index in block data

### ISSUE THREAD

**Issue by edd** [vegaprotocol/vega#9280]
### Problem encountered

https://github.com/vegaprotocol/frontend-monorepo/issues/4683

tldr the block explorer API provides the JSON for a transaction, decoded from the proto. There is also an index field. This index field sometimes matches the blockdata TX index. Sometimes it does not.

### Observed behaviour

On stagnet, for transaction `97A18A272F6157C76598DA23098CDF57DF3F0994EF7250CAE0310E73F72DB6C5`, the `index` of the transaction returned from the transaction data API is `0`.

Fetching the block from Tendermint and going in to the results and fetching tx 0 returns b64 that decodes to the same TX:
`curl 'https://be.stagnet1.vega.rocks/rest/transactions/97A18A272F6157C76598DA23098CDF57DF3F0994EF7250CAE0310E73F72DB6C5' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; rv:109.0) Gecko/20100101 Firefox/117.0' -H 'Accept: */*' -H 'Accept-Language: en-GB,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br' -H 'Referer: http://localhost:3000/' -H 'Origin: http://localhost:3000' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: cross-site' -H 'TE: trailers'`

However for later TXs in the block, this doesn't match. Initially I assumed this was something like the index in the TXdata didn't start at 0, but it doesn't appear to be a predictable off by one error.

### Expected behaviour

I have a way to fetch the b64 encoded protobuf to match the JSON

### Steps to reproduce

1. Go to https://explorer.fairground.wtf/txs/0x60770BA3D737FAAB37D8BC8AE1AE0618795FC38C2C2C6E49CE4B584ED3D69644
2. Open the TX JSON, see a `proposalSubmission`
3. Switch to base64, decode it, see a different TX
4. Check the API responses, see the index in the TX, see that TX index in the block, see that it does not match step 3

### Software version

0.72.x
